From 24ed8c20cd8088fb96b21868333b56571562befa Mon Sep 17 00:00:00 2001
From: Lennart Koschick <lennart@koschick.eu>
Date: Sun, 15 Jun 2025 07:27:50 +0000
Subject: [PATCH] Add S6 readiness notification

Upstream-Status: Inappropriate [S6 specific]
---
 src/svr-main.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/svr-main.c b/src/svr-main.c
index 6373e59..dc87258 100644
--- a/src/svr-main.c
+++ b/src/svr-main.c
@@ -204,6 +204,13 @@ static void main_noinetd(int argc, char ** argv, const char* multipath) {
 		fclose(pidfile);
 	}
 
+	const char *notification = getenv("NOTIFICATION_FD");
+	if (notification && *notification > '2' && *notification <= '9') {
+		int fd = *notification - '0';
+		(void)!write(fd, "\n", 1);
+		close(fd);
+	}
+
 	/* incoming connection select loop */
 	for(;;) {
 
